components/clarinet-files/src/project_manifest.rs

```diff
@@ -45,30 +46,161 @@ pub struct ProjectConfigFile {
     cache_dir: Option<String>,
 }
 
-#[derive(Serialize, Debug, Clone)]
+#[derive(Deserialize, Serialize, Debug, Clone)]
 pub struct ProjectManifest {
     pub project: ProjectConfig,
     #[serde(serialize_with = "toml::ser::tables_last")]
+    #[serde(deserialize_with = "contracts_deserializer")]
     pub contracts: BTreeMap<String, ClarityContract>,
     #[serde(rename = "repl")]
     pub repl_settings: repl::Settings,
     #[serde(skip_serializing)]
+    #[serde(default = "default_location")]
     pub location: FileLocation,
-    #[serde(skip_serializing)]
+    #[serde(skip_serializing, skip_deserializing)]
     pub contracts_settings: HashMap<FileLocation, ClarityContractMetadata>,
 }
 
-#[derive(Debug, Clone)]
+fn default_location() -> FileLocation {
+    let path = PathBuf::from_str("/tmp").expect("Unable to create a default location");
```

Comment:
I think this will fail for windows. I wonder if there's an easy way to get an OS-specific temp path?

Reply:
Done